File: VB/Form1.vb
Imports Microsoft.VisualBasic
Imports System
Imports System.Collections.Generic
Imports System.ComponentModel
Imports System.Data
Imports System.Drawing
Imports System.Linq
Imports System.Text
Imports System.Windows.Forms
Imports DevExpress.XtraNavBar.ViewInfo
Imports DevExpress.XtraNavBar
Imports System.Reflection
Imports DevExpress.XtraEditors.ViewInfo
Imports DevExpress.XtraEditors

Namespace WindowsFormsApplication44
	Partial Public Class Form1
		Inherits Form
		Public Sub New()
			InitializeComponent()

		End Sub
	End Class

	Public Class MyNavBarControl
		Inherits NavBarControl
		Public Sub New()
			MyBase.New()
		End Sub
		Protected Overrides Function CreateViewInfo() As NavBarViewInfo
			Return New MyExplorerBarNavBarViewInfo(Me)
		End Function
	End Class


	Public Class MyExplorerBarNavBarViewInfo
		Inherits ExplorerBarNavBarViewInfo
		Private Const currentNavBarScrollSpeed As Integer = 20

		Public Sub New(ByVal navBarControl As NavBarControl)
			MyBase.New(navBarControl)
		End Sub

		Public Overrides Property TopY() As Integer
			Get
				Return MyBase.TopY
			End Get
			Set(ByVal value As Integer)
				Dim isUp As Boolean = MyBase.TopY < value
				MyBase.TopY = If(isUp, value + currentNavBarScrollSpeed, value - currentNavBarScrollSpeed)
			End Set
		End Property
	End Class
End Namespace

File: CS/WindowsFormsApplication44/Form1.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Linq;
using System.Text;
using System.Windows.Forms;
using DevExpress.XtraNavBar.ViewInfo;
using DevExpress.XtraNavBar;
using System.Reflection;
using DevExpress.XtraEditors.ViewInfo;
using DevExpress.XtraEditors;

namespace WindowsFormsApplication44
{
    public partial class Form1 : Form
    {
        public Form1()
        {
            InitializeComponent();

        }
    }

    public class MyNavBarControl : NavBarControl
    {
        public MyNavBarControl() : base() { }
        protected override NavBarViewInfo CreateViewInfo()
        {
            return new MyExplorerBarNavBarViewInfo(this);
        }
    }


    public class MyExplorerBarNavBarViewInfo : ExplorerBarNavBarViewInfo
    {
        const int currentNavBarScrollSpeed = 20;
        
        public MyExplorerBarNavBarViewInfo(NavBarControl navBarControl) : base(navBarControl) { }

        public override int TopY
        {
            get
            {
                return base.TopY;
            }
            set
            {
                bool isUp = base.TopY < value;
                base.TopY = isUp ? value + currentNavBarScrollSpeed : value - currentNavBarScrollSpeed;
            }
        }
    }
}
